src/coreclr/System.Private.CoreLib/src/System/Reflection/MethodInvoker.CoreCLR.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Reflection
+{
+    internal partial class MethodInvoker
+    {
+        private readonly Signature _signature;
+        internal InvocationFlags _invocationFlags;
+
+        public MethodInvoker(MethodBase method, Signature signature)
+        {
+            _method = method;
+            _signature = signature;
+
+#if USE_NATIVE_INVOKE
+            // Always use the native invoke; useful for testing.
+            _strategyDetermined = true;
+#elif USE_EMIT_INVOKE
+            // Always use emit invoke (if IsDynamicCodeCompiled == true); useful for testing.
+            _invoked = true;
+#endif
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe object? InvokeNonEmitUnsafe(object? obj, IntPtr* arguments)
```

Comment:
Naming nits:
- I do not think we need to have `Unsafe` in all the method names. The fact that the method is unsafe is implied by the method signature. These are internal methods so we do not need to be highlighting that the method is unsafe.
- `InvokeNonEmit` -> `InterpretedInvoke`?